datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -1102,6 +1129,30 @@ where
         .collect()
 }
 
+pub(crate) fn get_mark_indices<T: ArrowPrimitiveType>(
+    range: &Range<usize>,
+    input_indices: &PrimitiveArray<T>,
+) -> PrimitiveArray<UInt32Type>
+where
+    NativeAdapter<T>: From<<T as ArrowPrimitiveType>::Native>,
+{
+    let mut bitmap = BooleanBufferBuilder::new(range.len());
+    bitmap.append_n(range.len(), false);
+    input_indices
+        .iter()
+        .flatten()
+        .map(|v| v.as_usize())
+        .filter(|v| range.contains(v))
+        .for_each(|v| {
+            bitmap.set_bit(v - range.start, true);
+        });
```

Comment:
I noticed that these lines are the same for `get_mark_indices`, `get_semi_indices` and `get_anti_indices`. We could factor them out to a new method?

Reply:
`map.filter.foreach` wondering can it be done in single iteration?